### PR TITLE
Fixes #822: only limit array size for indexed fields (ignore non-indexed)

### LINKED ISSUE #822: Document limits: only apply Max Array Element limit on indexed fields
As part of evaluation of Document limits (see DocumentLimitsConfig.java / CONFIGURATION.md), we want to:

* Change validation of "max-array-elements" to only cover indexed fields (that is, exclude no-index fields). This can be done by moving validation logic to occur after pruning no-index fields.

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Removes validation of max-array-size on non-indexed fields

**Which issue(s) this PR fixes**:
Fixes #822

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ public WritableShreddedDocument shred(JsonNode doc, UUID txId, DocumentProjector
     // Now that we have the traversable document, verify it does not violate
     // structural limits, before serializing.
     // (note: value validation has to wait until no-indexing projection is applied)
-    new StructuralValidator(documentLimits).validate(docWithId);
+    new FullDocValidator(documentLimits).validate(docWithId);
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java:78]: Changed name to better describe now that structural validation is split between full document (before trimming non-indexable fields) and indexable-only document.

File: src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java [hunk 2]
```diff
@@ -107,7 +107,7 @@ public WritableShreddedDocument shred(JsonNode doc, UUID txId, DocumentProjector
     }
 
     // and now we can finally validate (String) value lengths
-    new ValueValidator(documentLimits).validate(indexableDocument);
+    new IndexableValueValidator(documentLimits).validate(indexableDocument);
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java:110]: And similarly wrt indexable-only field validation.

File: src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java [hunk 3]
```diff
@@ -403,36 +384,61 @@ private void validateDocDepth(DocumentLimitsConfig limits, int depth) {
...
       this.limits = limits;
     }
 
     public void validate(ObjectNode doc) {
-      validateObjectValue(doc);
+      validateObjectValue(null, doc);
     }
 
-    private void validateValue(JsonNode value) {
+    private void validateValue(String referringPropertyName, JsonNode value) {
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java:402]: Need to pass name of property for value, to include in error messages (formerly not needed)

File: src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java [hunk 4]
```diff
@@ -286,30 +290,7 @@ private void validateArrayValue(
       ++depth;
       validateDocDepth(limits, depth);
 
-      if (arrayValue.size() > limits.maxArrayLength()) {
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java:289]: Moved out of full-doc validation, as it only should apply to indexable array fields.

**Review by Yuqi-Du — APPROVED** [review]
LGTM

**Pull request merged**